controllers/nutanixcluster_controller.go

```diff
@@ -333,9 +333,11 @@ func (r *NutanixClusterReconciler) reconcileCredentialRefDelete(ctx context.Cont
 	log := ctrl.LoggerFrom(ctx)
 	credentialRef, err := nutanixClient.GetCredentialRefForCluster(nutanixCluster)
 	if err != nil {
+		log.Error(err, fmt.Sprintf("error occurred while getting credential ref for cluster %s", nutanixCluster.Name))
```

Comment:
Please use structured logging (guidelines in https://github.com/kubernetes-sigs/controller-runtime/blob/v0.16.3/TMP-LOGGING.md?plain=1#L31 are useful btw).
```suggestion
		log = log.WithValues("cluster", client.ObjectKeyFromObject(cluster))
		log.Error(err, fmt.Sprintf("error occurred while getting credential ref for cluster")
```

My preference is to initialize the logger in the original `Reconcile` func like this to ensure all logs contain the object being reconciled:

```go
log := ctrl.LoggerFrom(ctx).WithValues("cluster", req.NamespacedName)
```

Reply:
@jimmidyson Structured logging is a great thing, but we need to change it not only in this line. Let's make logging improvement in separate PR.

Reply:
👍

Reply:
👍